### PR TITLE
Feature/test - Added Some Client-Side Tests For SignUpBox Component

### PR DESCRIPTION
**Pull request by willjw3**
Not all tests are able to be done with Enzyme, so I added the 'react-testing-library' dependency and used it for responsive tests.

### CODE REVIEW

**Comment by hitochan777** [comment]
@willjw3 I would like to hear your thoughts on how we organize test files for both server and client side. 
- One possibility is to put all test files right under `__test__`, but it will be hard to manage the test files as the number of files grow.
- The other possibility is to have a directory structure which is exactly the same as the test target, like
  ```
  __test__
      |__ server
      |__ client
               |__ page
                       |__ login.test.jsx
  ``` 

If we take the latter approach, I don't see a point of putting test files in `__test__` directory. Rather, we can put the test files together with the test target.

File: package.json [hunk 1]
```diff
@@ -59,10 +59,12 @@
...
     "eslint-plugin-react": "^7.12.4",
     "jest": "^24.5.0",
+    "jest-dom": "^3.1.3",
     "mongodb-memory-server": "^4.2.1",
     "nodemon": "^1.18.10",
     "pino-pretty": "^2.6.0",
     "prettier": "^1.16.4",
-    "react-test-renderer": "^16.8.5"
+    "react-test-renderer": "^16.8.5",
+    "react-testing-library": "^6.0.4"
```
Comment (hitochan777) [inline package.json:68]: I don't have deep understanding of testing libraries, but why do we have to use both `enzyme` and `react-testing-library`?

**Comment by willjw3** [comment]
According to React:

> Jest will look for test files with any of the following popular naming conventions:
> 
> Files with .js suffix in __tests__ folders.
> Files with .test.js suffix.
> Files with .spec.js suffix.
> The .test.js / .spec.js files (or the __tests__ folders) can be located at any depth under the src top level folder.
> 
> We recommend to put the test files (or __tests__ folders) next to the code they are testing so that relative imports appear shorter.

It seems we can create multiple __tests__ folders and place them near the files they are testing.

File: package.json (hunk 1 above)
Comment (willjw3) [reply package.json:68]: React Hooks are not supported in Enzyme yet, but they are in react-testing-library. However, I think Enzyme is easier to understand for testing component renderings. We could use only react-testing-library, but I don't see a problem with using them both. The important thing is that the tests run automatically and are written to isolate errors/failures well enough that the bad code can be quickly identified.

Comment (hitochan777) [reply package.json:68]: I see. Thank you for the info.

**Comment by hitochan777** [comment]
So what about placing test files in the same directory as the files they are testing, with the names ending with `.test.js` or `.spec.js`?

**Comment by willjw3** [comment]
That would be fine. Each import would have a nice, short relative path. Shall I set that up and re-request your review?

**Comment by hitochan777** [comment]
@willjw3 That would be nice. Sorry for the extra work.

File: client/components/SignUpBox.test.jsx [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+    // input form fields are completely empty
+    expect(wrapper.find("p").text()).toEqual(
+      "Password must be at least 6 characters in length"
+    );
+  });
+});
+
+test("SignUpBox loads with intitial state of '' ", () => {
+  const { container } = render(<SignUpBox />);
+  const emailVal = getByTestId(container, "emailVal");
```
Comment (hitochan777) [inline client/components/SignUpBox.test.jsx:65]: Can you get `emailVal` by just calling `.find("#email")`, instead of using `getByTestId`?

Comment (willjw3) [reply client/components/SignUpBox.test.jsx:65]: No, you would have to use `const emailVal = mount(<SignUpBox />).find('#email')`. That's the Enzyme way. I used the `data-testId` to keep it unique to react-testing-library.

Comment (hitochan777) [reply client/components/SignUpBox.test.jsx:65]: Ah I see. `react-testing-library` does not have an equivalent of `.find` in enzyme.

**Review by hitochan777 — APPROVED** [review]

**Comment by hitochan777** [comment]
LGTM!

**Pull request merged**